app/controllers/users/two_factor_authentication_setup_controller.rb

```diff
@@ -20,15 +20,24 @@ def create
       if result.success?
         process_valid_form
       else
-        @presenter = two_factor_options_presenter
-        render :index
+        direct_user_on_failure
       end
     end
 
     private
 
+    def direct_user_on_failure
+      if session[:signing_up] && @two_factor_options_form.selection == 'backup_code_only'
+        session[:signing_up] = false
```

Comment:
This needs coverage

Reply:
Refactored and covered by the feature spec